app/uk/gov/hmrc/gform/gform/GformSubmission.scala

```diff
@@ -34,21 +34,23 @@ object GformSubmission {
     gformConnector: GformConnector,
     retrievals: MaterialisedRetrievals,
     formTemplate: FormTemplate,
+    emailParameters: EmailParametersRecalculated,
     maybeAccessCode: Option[AccessCode],
     customerId: CustomerId,
     htmlForPDF: String
   )(implicit hc: HeaderCarrier): Future[HttpResponse] =
     gformConnector.submitFormWithPdf(
       FormId(retrievals, formTemplate._id, maybeAccessCode),
       customerId,
-      buildSubmissionData(htmlForPDF, customerId, retrievals, formTemplate),
+      buildSubmissionData(htmlForPDF, customerId, retrievals, formTemplate, emailParameters),
       AffinityGroupUtil.fromRetrievals(retrievals)
     )
 
   private def buildSubmissionData(
     htmlForPDF: String,
     customerId: CustomerId,
     retrievals: MaterialisedRetrievals,
-    formTemplate: FormTemplate): SubmissionData =
-    SubmissionData(htmlForPDF, VariablesBuilder(retrievals, formTemplate, customerId))
+    formTemplate: FormTemplate,
+    emailParameters: EmailParametersRecalculated): SubmissionData =
+    SubmissionData(htmlForPDF, VariablesBuilder(retrievals, formTemplate, customerId, emailParameters))
```

Comment:
Putting `emailParameters` into `VariablesBuilder` is just complicating stuff since `VariablesBuilder` is supposed to be for variables in handlebar templates.

Let's put `emailParameters` into it's own (new) field in `SubmissionData`.